### PR TITLE
docs(website): display sidebar rules on tombstone pages

### LINKED ISSUE #9213: Website: Rules sidebar missing from "tombstone" pages
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Rules pages on the site normally have a left page sidebar based on https://github.com/typescript-eslint/typescript-eslint/blob/50cc96dc0bd5f8c6b4501c7c277365226be08095/packages/website/sidebars/sidebar.rules.js. But, that's not there at the moment for pages like `camelcase.md` that don't have an associated rule:

Screenshot of https://typescript-eslint.io/rules/camelcase showing no left sidebar

### Affected URL(s)

https://typescript-eslint.io/rules/camelcase & more

💖

### PR DESCRIPTION
**Pull request by y-hsgw**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9213 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Fixed the sidebar to display on tombstone pages.

ex: camelcase
スクリーンショット 2024-08-06 15 14 59

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
Oh wow I'd forgotten we'd called it `someSidebar` two years ago. Amusing. 😂 

Thanks for the fix, this is helpful!

File: packages/eslint-plugin/docs/rules/ban-types.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+---
+displayed_sidebar: someSidebar
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/ban-types.md:2]: I suppose we should probably rename it to `rulesSidebar` in `sidebar.rules.js`. But that can be a followup issue after this PR IMO.

Comment (Josh-Cena) [reply packages/eslint-plugin/docs/rules/ban-types.md:2]: Let's do that as part of this PR.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
(requesting the name change)

File: packages/eslint-plugin/docs/rules/ban-types.md (hunk 1 above)
Comment (y-hsgw) [reply packages/eslint-plugin/docs/rules/ban-types.md]: Renamed to rulesSidebar.
dd2ff2d

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by Josh-Cena — APPROVED** [review]
Thanks!

**Pull request merged**